h2o-algos/src/main/java/hex/anovaglm/AnovaGLMUtils.java

```diff
@@ -0,0 +1,454 @@
+   * @param dinfo
+   */
+  public static void generatePredictorNames(String[][] predComboNames, String[][] predictorNames, int[] predColumnStart, 
+                                            int[] degreeOfFreedom, DataInfo dinfo) {
+    int predNums = predComboNames.length;
+    int colStart = 0;
+    for (int predInd = 0; predInd < predNums; predInd++) {
+      if (predComboNames[predInd].length == 1) {
+        if (dinfo._adaptedFrame.vec(predComboNames[predInd][0]).domain() == null) // one numeric column
+          predictorNames[predInd] = new String[]{predComboNames[predInd][0]};
+        else 
+          predictorNames[predInd] = transformOneCol(dinfo._adaptedFrame, predComboNames[predInd][0]);
+      } else {  // working with interaction columns
+          predictorNames[predInd] = transformMultipleCols(dinfo._adaptedFrame, predComboNames, predInd, predictorNames);
+      }
+      colStart = updateDOFColInfo(predInd, predictorNames[predInd], degreeOfFreedom, predColumnStart, colStart);
+    }
+  }
+  
+  public static int updateDOFColInfo(int predInd, String[] predComboNames, int[] dof, int[] predCS, int offset) {
+    dof[predInd] = predComboNames.length;;
+    predCS[predInd] = offset;
+    return dof[predInd]+offset;
+  }
+  
+  public static int findComboMatch(String[][] predComboNames, int currIndex) {
+    String[] currCombo = predComboNames[currIndex];
+    int startPos = 1;
+    for (int comboSize = currCombo.length-1; comboSize >= 0; comboSize--) {
+      String[] smallerCurrCombo = Arrays.copyOfRange(currCombo, startPos++, currCombo.length);
+      for (int sInd = currIndex - 1; sInd >= 0; sInd--) {
+        if (Arrays.equals(smallerCurrCombo, predComboNames[sInd]))
+          return sInd;
+      }
+    }
+    return -1;
+  }
+
+  public static String[] combineAndFlat(String[][] predictComboNames) {
+    int numCombos = predictComboNames.length;
+    String[] finalPredNames = new String[numCombos];
+    
+    for (int index = 0; index < numCombos; index++) {
+      String start = predictComboNames[index][0];
+      if (predictComboNames[index].length > 1)
+      for (int subIndex = 1; subIndex < predictComboNames[index].length; subIndex++)
+        start = start +":"+predictComboNames[index][subIndex];
+      finalPredNames[index] = start;
+    }
+    return finalPredNames;
+  }
+
+  public static String[] transformMultipleCols(Frame vec2Transform, String[][] predComboNames, int currIndex,
+                                               String[][] predNames) {
+    String[] currPredCombo = predComboNames[currIndex];
+    int matchPreviousCombo = findComboMatch(predComboNames, currIndex);
+    String[] matchPredCombo = predComboNames[matchPreviousCombo];
+    String[] matchPredNames = predNames[matchPreviousCombo];
+    String[] searchPair = new String[]{currPredCombo[0], currPredCombo[1]};
+    matchPredNames = transformTwoCols(vec2Transform, searchPair, matchPredNames);
```

Comment:
I would prefer to return the value immediately instead of rewriting existing local variable and returning it

Reply:
I fixed it!  Thanks!